packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -58,7 +58,7 @@ export default (clipanion: any, pluginConfiguration: PluginConfiguration) => cli
       const configuration = await Configuration.find(cwd, pluginConfiguration);
       const {project} = await Project.find(configuration, cwd);
 
-      let workspaces = project.workspaces.filter(workspace => workspace.manifest.scripts.has(command));
+      let workspaces = project.workspaces.filter(workspace => workspace.manifest.scripts.has(command) && (command !== process.env.npm_lifecycle_event || workspace.cwd !== cwd));
```

Comment:
```suggestion
      // Prevents infinite loop in the case of configuring a script as such:
      //     "lint": "yarn workspaces foreach run lint"
      const isRecursing = command !== process.env.npm_lifecycle_event;

      let workspaces = project.workspaces.filter(workspace => workspace.manifest.scripts.has(command) && (!isRecursing || workspace.cwd !== cwd));
```

Reply:
Shouldn't that be

```ts
const isRecursing = command === process.env.npm_lifecycle_event;
```

?

Reply:
Woopsie yeah, I forgot to revert the condition 😅